corehq/apps/cloudcare/views.py

```diff
@@ -589,7 +589,15 @@ def _fail(error):
 
     build = _fetch_build(domain, request.couch_user.username, app_id)
     if not build:
-        return _fail(_("Could not find application."))
+        # These links can be used for cross-domain web apps workflows, where a link jumps to the
```

Comment:
@orangejenny I am having little trouble contextualizing this. In particular what are upstream and downstream apps and how they are different from source and traget apps here ?

Reply:
I meant that "source" is the app that the user is currently looking at in web apps, while "target" is the app that they're going to. Both the source and target apps are linked/downstream apps of the same upstream app.

So you're in county A, looking at patients from counties A, B, C, and D. If you click on a patient in B, C, or D, you need a smart link. I'd like to use one smart link template for all of those patients, something like `/a/{DOMAIN}/app/v1/{APP_ID}/endpoint_id`. Web apps / formplayer can replace the domain in the URL, since the case will know what domain it belongs to (thanks to https://github.com/dimagi/commcare-hq/pull/30388), but it doesn't know what app id to use. I'm using the upstream app id in the URL and then having HQ look up the actual app to redirect to.

Reply:
makes sense, thanks for the detailed explanation.